src/webui/src/components/trial-detail/Intermeidate.tsx

```diff
@@ -204,28 +213,73 @@ class Intermediate extends React.Component<IntermediateProps, IntermediateState>
         this.drawIntermediate(source);
     }
 
-    componentWillReceiveProps(nextProps: IntermediateProps) {
-        const { isFilter, filterSource } = this.state;
-        if (isFilter === true) {
-            const pointVal = this.pointInput !== null ? this.pointInput.value : '';
-            const minVal = this.minValInput !== null ? this.minValInput.value : '';
-            if (pointVal === '' && minVal === '') {
-                this.drawIntermediate(nextProps.source);
+    componentWillReceiveProps(nextProps: IntermediateProps, nextState: IntermediateState) {
+        const { isFilter, filterSource } = nextState;
+        const { whichGraph, source } = nextProps;
+        
+        if (whichGraph === '4') {
```

Comment:
what does '4' mean in this context?

Reply:
'4' means that user is looking at intermediate result graph. I'll just render intermediate graph rather then all graphs.